open-banking-brasil-financial-api-1_ID3.md

```diff
@@ -412,7 +416,8 @@ In addition to the requirements outlined in Open Banking Brasil security provisi
 7. shall retain a complete audit history of the consent resource in accordance with current Central Bank brazilian regulation;
 8. shall return authentication failure and return code _access_denied_ in the _error_ parameter (as specified in section 4.1.2.1 of [RFC6749]) if the CPF of the authenticated user is not the same as indicated in the _loggedUser_ element of the Consent Resource Object;
 9. shall return authentication failure and return code _access_denied_ in the _error_ parameter (as specified in section 4.1.2.1 of [RFC6749]) if the _businessEntity_ element has not been populated in the related Consent Resource Object and the user has selected or authenticated by using a credential related to a business account;
-10. an autenticated or selected business account´s CNPJ must match the value present in the _businessEntity_ element of the Consent Resource Object. In case of divergence authorization server shall return authentication failure and return code _access_denied_ in the _error_ parameter (as specified in section 4.1.2.1 of [RFC6749]).
+10. an autenticated or selected business account´s CNPJ must match the value present in the _businessEntity_ element of the Consent Resource Object. In case of divergence authorization server shall return authentication failure and return code _access_denied_ in the _error_ parameter (as specified in section 4.1.2.1 of [RFC6749]);
+11. an issued _refresh_token_ shall be valid until the linked Consent Resource object is still valid.
```

Comment:
Great! This was contentious but fully supportive of including this back in.

Reply:
Some banks was defining an refresh smaller than consent... this way we should define how refresh should be "refreshed" (one time refresh so on..) this seams to be the easy way to drive this "problem"...

Reply:
Sorry i'm confused. Do you meant that banks were issuing refresh tokens that would expire before the consent record would expire? That's amusing however yes the specification did not explicitly say that the grant had to be issued for the consent window. I though it was obvious. May i suggest the language

Can i suggest the following language

shall ensure refresh tokens expiration time is at least equal to the linked consent resource expiration time.

Reply:
It shall be exactly the same expiration time... it cannot be greater than consent expiration time, right?

Reply:
So, I originally had this exact wording but it was rejected by the security group as some vendors could not tie the refresh token / grant life time to the underlying consent life time. I would prefer to be explicitly tied to the underlying consent however I believe the consensus was for the refresh token expiry to be “at least as long” as the underlying consent.